embedded-graphics/src/pixelcolor/mod.rs

```diff
@@ -1,57 +1,443 @@
+
+    /// White color (Y = 100%)
+    pub const WHITE: Y8 = Self::new(255);
+}
+
+impl PixelColor for Y8 {
+    const DEFAULT_BG: Self = Y8::BLACK;
+    const DEFAULT_FG: Self = Y8::WHITE;
+}
+
+impl FromSlice for Y8 {
+    fn from_be_slice(data: &[u8]) -> Self {
+        Self(data[0])
+    }
+
+    fn from_le_slice(data: &[u8]) -> Self {
+        Self(data[0])
+    }
+}
+
+/// Binary color
 ///
-/// This trait should be implemented for any type an embedded graphics item can be rendered into.
-/// It is implemented for `u8`, `u16` and `u32` so any display drivers that use these types for
-/// their pixel values should work out of the box.
-///
-/// If a custom pixel type is used in a driver, `PixelColor` must be implemented for it to be
-/// compatible. Below is a contrived example that wraps a `u16` in a struct.
-///
-/// ```rust
-/// use embedded_graphics::drawable::Pixel;
-/// use embedded_graphics::unsignedcoord::UnsignedCoord;
-/// use embedded_graphics::pixelcolor::PixelColor;
-///
-/// // `Copy` and `Clone` are bounds on `PixelColor` and are required.
-/// // `PartialEq` is for the `assert_eq!()` later in this example.
-/// // `Debug` is for convenience :)
-/// #[derive(Copy, Clone, PartialEq, Debug)]
-/// struct CustomPixelColor {
-///     pub value: u16,
-/// }
-///
-/// impl CustomPixelColor {
-///     fn new(color: u16) -> Self {
-///         CustomPixelColor { value: color }
-///     }
-/// }
-///
-/// impl PixelColor for CustomPixelColor {}
-///
-/// // `From<u8>` is a bound on `PixelColor` so must be implemented for your pixel colour type. You
-/// // can also implement `From` for other types like `u16`, etc for convenience.
-/// impl From<u8> for CustomPixelColor {
-///     fn from(other: u8) -> Self {
-///         CustomPixelColor {
-///             value: other as u16,
-///         }
-///     }
-/// }
-///
-/// fn main() {
-///     let colour = CustomPixelColor::new(127u16);
-///
-///     assert_eq!(colour.value, 127u16);
-/// }
-/// ```
-pub trait PixelColor: Clone + Copy + From<u8> {}
-
-impl PixelColor for u8 {}
-impl PixelColor for u16 {}
-impl PixelColor for u32 {}
+/// `BinaryColor` is used for displays and images with two possible states.
+/// The interpretation of active and inactive states can vary for different uses and isn't specified.
+/// `BinaryColor::On` might represent a black pixel on a LCD panel and a white pixel on an OLED display.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum BinaryColor {
+    /// An inactive pixel.
+    Off,
+
+    /// An active pixel.
+    On,
+}
+
+impl BinaryColor {
+    /// Inverts the color.
+    pub fn invert(&self) -> Self {
+        match self {
+            BinaryColor::On => BinaryColor::Off,
+            BinaryColor::Off => BinaryColor::On,
+        }
+    }
+
+    /// Maps active and inactive colors to a different `PixelColor` type.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use embedded_graphics::pixelcolor::{BinaryColor, RgbColor, Rgb565};
```

Comment:
Nit: I think it's useful for examples to show where the items used are imported from

```suggestion
    /// use embedded_graphics::pixelcolor::{BinaryColor, RgbColor, Rgb565};
```